java/demo/src/com/google/phonenumbers/PhoneNumberParserServlet.java

```diff
@@ -103,13 +103,46 @@ public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOEx
     resp.setContentType("text/html");
     resp.setCharacterEncoding(UTF_8.name());
     if (fileContents.length() == 0) {
-      resp.getWriter().println(
-          getOutputForSingleNumber(phoneNumber, defaultCountry, languageCode, regionCode));
+      // Redirect to a URL with the given input encoded in the query parameters.
+      try {
+        resp.sendRedirect(getPermaLinkURL(phoneNumber, defaultCountry, languageCode, regionCode));
+      } catch(UnsupportedEncodingException e) {
+        resp.getWriter().println(StringEscapeUtils.escapeHtml(e.toString()));
```

Comment:
This is unreachable code (UTF-8 is guaranteed in Java). You should just add a comment here and make this throw:
  new AssertionError(e);